### PR TITLE
Revert using clang-format in the examples

### PR DESCRIPTION
**Pull request by masterleinad**
I hear agreement (https://github.com/dealii/dealii/issues/6686#issuecomment-392490064) that we should not use `clang-format` for the examples.
This PR reverts applying clang-format keeping all the other changes since then.

### CODE REVIEW

**Comment by kronbichler** [comment]
Obviously I agree on this change, but let us hear what the others say.

**Comment by kronbichler** [comment]
Or maybe to expand a bit: Is there a subset of rules that we could define that does stop some of the more aggressive formatting but does some basic stuff, like the spaces in `for (unsigned int i = 0; i < n; ++i)`? Is that something that we could agree on (assuming that it is technically feasible, I'm not into that part)?

**Comment by masterleinad** [comment]
No, we can't do just that with `clang-format`. As @tamiko said multiple times, `clang-format` creates an AST it then parses according to the given formatting options. We can't just keep all line breaks or similar.

**Comment by bangerth** [comment]
Hang on for now -- I've got a longer discussion about this coming.

**Comment by tamiko** [comment]
I agree. Let's hold horses here for a moment.

File: examples/step-5/step-5.cc [hunk 1]
```diff
@@ -164,69 +167,75 @@ void Step5<dim>::setup_system()
...
-                cell_matrix(i, j) +=
-                  (current_coefficient * fe_values.shape_grad(i, q_index) *
-                   fe_values.shape_grad(j, q_index) * fe_values.JxW(q_index));
-
-              cell_rhs(i) += (fe_values.shape_value(i, q_index) * 1.0 *
+              for (unsigned int j=0; j<dofs_per_cell; ++j)
+                cell_matrix(i,j) += (current_coefficient *
+                                     fe_values.shape_grad(i,q_index) *
+                                     fe_values.shape_grad(j,q_index) *
+                                     fe_values.JxW(q_index));
```
Comment (tamiko) [inline examples/step-5/step-5.cc]: About the only places that I can see and where I absolutely agree with @bangerth that they are reindented suboptimally are these long mathematical expressions of bilinear forms (well and some awkward operator output). As I said, we can instruct clang format to not apply line breaks here. There are two possibilities to achieve this:
```
cell_matrix(i, j) += (current_coefficient *              //
                      fe_values.shape_grad(i, q_index) * //
                      fe_values.shape_grad(j, q_index) * //
                      fe_values.JxW(q_index));           //
```
```
/* clang-format off */
cell_matrix(i, j) += (current_coefficient *
                      fe_values.shape_grad(i, q_index) *
                      fe_values.shape_grad(j, q_index) *
                      fe_values.JxW(q_index));
/* clang-format on */
```
I actually like the first variant. There, clang format will still reindent everything but simply keeps the line breaks.

File: examples/step-10/step-10.cc [hunk 2]
```diff
@@ -131,22 +131,24 @@ namespace Step10
...
             grid_out.set_flags(gnuplot_flags);
 
             // Finally, generate a filename and a file for output:
-            std::string filename = filename_base + "_mapping_q_" +
-                                   Utilities::to_string(degree) + ".dat";
-            std::ofstream gnuplot_file(filename);
+            std::string filename = filename_base
+                                   + "_mapping_q_"
+                                   + Utilities::to_string(degree)
+                                   + ".dat";
```
Comment (tamiko) [inline examples/step-10/step-10.cc]: That's another instance, where not smuching everything into as few lines as possible looks better:
```
std::string filename = filename_base                  //
                       + "_mapping_q_"                //
                       + Utilities::to_string(degree) //
                       + ".dat";                      //
```

File: examples/step-5/step-5.cc (hunk 1 above)
Comment (kronbichler) [reply examples/step-5/step-5.cc]: I have commented more about this over here https://github.com/tamiko/dealii/pull/10#discussion_r191663138
I think that the first variant is nice as well. I'm a bit undecided whether we would like to remove those trailing `//` characters (or `clang-format on/off`) as part of the process generating the online documentation, but in the end I'm fine to accept the variant that the majority agrees upon.

Comment (bangerth) [reply examples/step-5/step-5.cc]: We could use the trailing slashes to explain again what each part of the formula means. For everything else, remember that people eventually look at the code -- they will ask themselves and the instructor what exactly any of our markup means. We could lessen the impact by processing the source files in some way before we install them, but that's suboptimal. I think that leaving some macro in is not a good option.

If we want to go with any kind of mark-up, there are two questions:
* Who is going to do the work? We have ~60 tutorial programs, so someone will have to find 100-150 places that need to be fixed. That'll take several hours.
* What this suggestion really is is to work around the limitations of a tool. That's annoying -- do we really want to commit to this?

Comment (tamiko) [reply examples/step-5/step-5.cc]: @bangerth We will have to go through all examples in the next weeks anyway. This includes
 - necessary changes and documentation updates for `AffineConstraints` (once merged)
 - C++11 improvements

We could add markup fixes to the list. With 4 people working on it it is also only 15 example steps per person. (Preferably one pull request per example step and a list to tick off) That's doable over a course of a month. (This implies we trust each other to a certain extent with respect to the markup changes.)

**Comment by tamiko** [comment]
Let's close this PR and revisit it later.

**Pull request closed without merging**